### PR TITLE
Add Support for Time-Enabled Layers in Cesium

### PR DESCRIPTION
**Pull request by htran84**
Add support for specifying a list of time steps in the MVLayer to be used to implement animations for time-enabled layers.
Implemented support in the CesiumMapView for now. Not supported in the main MapView as of yet.

### CODE REVIEW

File: tethys_gizmos/gizmo_options/map_view.py [hunk 1]
```diff
@@ -525,7 +525,7 @@ class MVLayer(SecondaryGizmoOptions):
         legend_extent (list): A list of four ordinates representing the extent that will be used on "zoom to layer": [minx, miny, maxx, maxy].
         legend_extent_projection (str): The EPSG projection of the extent coordinates. Defaults to "EPSG:4326".
         data (dict): Dictionary representation of layer data
-
+        times (list): only for Cesium. List of start time for geoserver image mosaic layers. (e.g.: ["20210322T112511Z", "20210322T122511Z", "20210322T132511Z"])
```
Comment (swainn) [inline tethys_gizmos/gizmo_options/map_view.py:528]: Change to: 

```
times (list): List of time steps if layer is time-enabled. Times should be represented as strings in ISO 8601 format (e.g.: ["20210322T112511Z", "20210322T122511Z", "20210322T132511Z"]). Currently only supported in CesiumMapView.
```

File: tethys_gizmos/static/tethys_gizmos/js/cesium_map_view.js [hunk 2]
```diff
@@ -783,6 +808,12 @@ var CESIUM_MAP_VIEW = (function() {
         return true;
     }
 
+    cesium_time_callback = function (interval) {
```
Comment (swainn) [inline tethys_gizmos/static/tethys_gizmos/js/cesium_map_view.js:811]: Remove space between `function` and call arguments: `cesium_time_callback = function(interval) {`

File: tethys_gizmos/static/tethys_gizmos/js/cesium_map_view.js [hunk 3]
```diff
@@ -195,12 +195,37 @@ var CESIUM_MAP_VIEW = (function() {
...
                         parameters.VIEWPARAMS = curr_layer.options.params.VIEWPARAMS;
                     }
-
-                    var tile_wms = new Cesium.WebMapServiceImageryProvider({
-                        url: curr_layer.options.url,
-                        layers: curr_layer.options.params.LAYERS,
-                        parameters: parameters
-                    });
+                    if (curr_layer.times) {
+                        // times should be in this format "["20210322T112511Z", "20210322T122511Z", "20210323T032511Z"]"
```
Comment (swainn) [inline tethys_gizmos/static/tethys_gizmos/js/cesium_map_view.js:199]: `// times should a JSON array string with times in ISO 8601 format: "["20210322T112511Z", "20210322T122511Z", "20210323T032511Z"]"`

File: tethys_gizmos/static/tethys_gizmos/js/cesium_map_view.js [hunk 4]
```diff
@@ -195,12 +195,37 @@ var CESIUM_MAP_VIEW = (function() {
...
+                        var times = JSON.parse(curr_layer.times);
+                        const provider_interval = new Cesium.TimeIntervalCollection.fromIso8601DateArray({
+                            iso8601Dates: JSON.parse(curr_layer.times),
```
Comment (swainn) [inline tethys_gizmos/static/tethys_gizmos/js/cesium_map_view.js:202]: why not use the `times` variable here (see line 200)?

File: tethys_gizmos/static/tethys_gizmos/js/cesium_map_view.js [hunk 5]
```diff
@@ -195,12 +195,37 @@ var CESIUM_MAP_VIEW = (function() {
...
+                            dataCallback: cesium_time_callback,
+                        });
+
+                        var clock = m_viewer.clock;
+                        var start = Cesium.JulianDate.fromIso8601(times[0])
+                        var stop = Cesium.JulianDate.fromIso8601(times[times.length - 1])
+                        clock.startTime = start;
+                        clock.stopTime = stop;
+                        clock.currentTime = start;
+                        clock.multiplier = 600;
```
Comment (swainn) [inline tethys_gizmos/static/tethys_gizmos/js/cesium_map_view.js:212]: Why 600?

**Review by swainn — CHANGES_REQUESTED** [review]

**Review by swainn — APPROVED** [review]

**Review by sdc50 — APPROVED** [review]

**Pull request merged**